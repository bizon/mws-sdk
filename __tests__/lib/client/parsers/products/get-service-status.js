const parseXml = require('../../../../../lib/client/parsers')
const parseGetServiceStatusResponse = require('../../../../../lib/client/parsers/base/get-service-status-response')

describe('lib.client.parsers.base.get-service-status-response', () => {
  it('should parse the GetServiceStatusResponse example response from MWS doc', () => {
    const doc = parseXml(
      `<?xml version="1.0"?>
      <GetServiceStatusResponse
          xmlns="http://mws.amazonservices.com/schema/Products/2011-10-01">
          <GetServiceStatusResult>
              <Status>GREEN</Status>
              <Timestamp>2012-12-04T22:53:58.830Z</Timestamp>
          </GetServiceStatusResult>
          <ResponseMetadata>
              <RequestId>0a51dc17-20bf-4e0d-8ac5-5d53bd1129cf</RequestId>
          </ResponseMetadata>
      </GetServiceStatusResponse>`,
    )

    const response = parseGetServiceStatusResponse('/products:GetServiceStatusResponse', doc)

    expect(response).toMatchSnapshot()
  })
})
