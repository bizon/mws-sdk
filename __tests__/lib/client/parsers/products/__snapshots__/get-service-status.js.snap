// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib.client.parsers.base.get-service-status-response should parse the GetServiceStatusResponse example response from MWS doc 1`] = `
Object {
  "getServiceStatusResult": Object {
    "status": "GREEN",
    "timestamp": 2012-12-04T22:53:58.830Z,
  },
  "responseMetadata": Object {
    "requestId": "0a51dc17-20bf-4e0d-8ac5-5d53bd1129cf",
  },
}
`;
